src/armature/GuidingVectors.ts

```diff
@@ -183,45 +191,147 @@ export class GuidingVectors implements CostFn {
+     * attached to.
+     * @returns {number} The cost, given the above.
+     */
+    private computeCost(closest: Closest, added: vec4, parentPosition: vec3): number {
+        const distance = vec3.distance(Mapper.coordToVector(<coord>closest.point), parentPosition);
+
+        // Evaluate distance cost polynomial using Horner's method
+        let distanceCost = 0;
+        for (let power = 2; power >= 0; power -= 1) {
+            distanceCost = closest.curve.distanceMultiplier[power] + distanceCost * distance;
+        }
+
+        // Add cost for vector alignment
+        const alignmentCost =
+            (-vec3.dot(closest.guidingVector, vec3From4(added)) + closest.curve.alignmentOffset) *
+            closest.curve.alignmentMultiplier;
+
+        return alignmentCost + distanceCost;
+    }
+
+    /**
+     * Given a spawn point, computes and stores the closest point on the guiding curves for
+     * that point.
+     */
+    private getOrCreateSpawnPointClosest(spawnPoint: SpawnPoint): Closest {
+        let closest = this.spawnPointClosests.get(spawnPoint);
+
+        if (closest === undefined) {
+            const localToGlobalTransform = spawnPoint.at.node.localToGlobalTransform();
+            const point = vec4.fromValues(0, 0, 0, 1);
+            vec4.transformMat4(point, point, localToGlobalTransform);
+
+            closest = this.closest(point);
+
+            this.spawnPointClosests.set(spawnPoint, closest);
+        }
+
+        return closest;
+    }
 
-            // Compare the new structure's vector with the direction vector for the curve point
-            const guidingVector = Mapper.coordToVector(<coord>closest.curve.bezier.derivative(
-                <number>closest.point.t
-            ));
-            vec3.normalize(guidingVector, guidingVector);
+    /**
+     * Given a spawn point and a generator, computes the average direction vector coming from that
+     * spawn point.
+     */
+    private getOrCreateSpawnPointVector(generator: Generator, spawnPoint: SpawnPoint): vec4 {
+        let vector = this.spawnPointVectors.get(spawnPoint);
 
-            const alignmentCost =
-                (-vec3.dot(guidingVector, vec3From4(vector)) + closest.curve.alignmentOffset) *
-                closest.curve.alignmentMultiplier;
+        if (vector === undefined) {
+            const expectedVector = this.getOrCreateExpectedVector(generator, spawnPoint.component);
+            const localToGlobalTransform = spawnPoint.at.node.localToGlobalTransform();
+            vector = vec4.transformMat4(vec4.create(), expectedVector, localToGlobalTransform);
 
-            // Add cost for the distance away from the curve
-            const closestPoint = vec3ToVector(Mapper.coordToVector(<coord>closest.point));
-            const distance = vec4.distance(closestPoint, parentPosition);
+            // To avoid doing a matrix multiplication to orient the direction vector for a component
```

Comment:
This comment is confusingly worded, especially when it just describes the work that the reader just read in the code. Just saying something like "Cache the spawn point direction vector" should be good enough.

Reply:
It is intentional, so I'll update the comment